src/ncp/rcp_host.hpp

```diff
@@ -60,11 +60,34 @@ class FeatureFlagList;
 
 namespace Ncp {
 
+/**
+ * This class implements the NetworkProperties under RCP mode.
+ *
+ */
+class RcpNetworkProperties : virtual public NetworkProperties
+{
+public:
+    /**
+     * Constructor.
+     *
+     */
+    explicit RcpNetworkProperties(void);
+
+    // NetworkProperties methods
+    otDeviceRole GetDeviceRole(void) const override;
+
+    // Set the otInstance
+    void SetInstance(otInstance *aInstance);
+
+private:
+    otInstance *mInstance;
+};
```

Comment:
This is applicable to RCP but seems to be even more general-purpose. 

I think it can be used whenever the OpenThread stack itself is directly accessible (i.e., we can call OpenThread APIs directly). Should we perhaps rename/define it to indicate such a model?

This model might be helpful in other (future) architectures we could support, e.g., an architecture where the entire OpenThread stack, along with the radio driver and platform, is part of the same firmware (running on the same MCU).

Reply:
SG! I think of a few namings. It should reflect that "OpenThread APIs is directly accessible". What about using `OtNetworkProperties`?

Reply:
Yes. Sounds good to me. 👍

Reply:
Updated.